### PR TITLE
[FLINK-15998] In Glossary, separate application/job cluster

### PR DESCRIPTION
**Pull request by aljoscha**
@knaufk Could you please review?

cc @morsapaes

### CODE REVIEW

File: docs/concepts/glossary.md [hunk 1]
```diff
@@ -25,11 +25,16 @@ under the License.
 
 #### Flink Application Cluster
 
-A Flink Application Cluster is a dedicated [Flink Cluster](#flink-cluster) that only
+A Flink Application Cluster is a [Flink Cluster](#flink-cluster) that
```
Comment (knaufk) [inline docs/concepts/glossary.md]: What is a "Flink Application" vs a "Flink Job"?

**Comment by aljoscha** [comment]
Ah, this is a bit more difficult than I though, because the glossary often refers to "Flink Jobs" as the runtime representation of "Flink Programs", but a "Flink Application" should really be the "program", and that can spawn multiple jobs. Let me think about it, I might not finish it now because I'm on vacation starting today for three weeks. 😅

**Comment by aljoscha** [comment]
@knaufk Let's try this a second time. 😅

**Comment by aljoscha** [comment]
Thanks for the review!

**Pull request closed without merging**